modules/extension/app-schema/app-schema/src/test/java/org/geotools/appschema/jdbc/JoiningJDBCFeatureSourceTest.java

```diff
@@ -60,4 +62,26 @@ public void testMultipleIds() throws Exception {
         assertEquals("FOREIGN_ID_0_0", type.getDescriptor(0).getName().getLocalPart());
         assertEquals("FOREIGN_ID_0_1", type.getDescriptor(1).getName().getLocalPart());
     }
+
+    @Test
+    public void testSimpleFeatureTypeNoPk() throws IOException {
+        SimpleFeatureTypeBuilder builder = new SimpleFeatureTypeBuilder();
+        builder.setName("test");
+
+        JoiningQuery query = new JoiningQuery();
+        JoiningQuery.QueryJoin join = new JoiningQuery.QueryJoin();
+        join.getIds().add("one");
+        join.getIds().add("two");
+        query.setQueryJoins(Collections.singletonList(join));
+
+        JDBCDataStore mockStore = Mockito.mock(JDBCDataStore.class);
+        ContentEntry mockEntry = Mockito.mock(ContentEntry.class);
+
+        Mockito.when(mockStore.getSQLDialect()).thenReturn(new PostGISDialect(mockStore));
+        Mockito.when(mockEntry.getDataStore()).thenReturn(mockStore);
+        JoiningJDBCFeatureSource source =
+                new JoiningJDBCFeatureSource(new JDBCFeatureSource(mockEntry, null));
+
+        assertNull(source.getPrimaryKey());
```

Comment:
seems to not test the code path that has been modified. Probably the getCountInternal method should be tested instead.